tests/queries/0_stateless/02710_allow_suspicious_indices.sql

```diff
@@ -0,0 +1,19 @@
+DROP TABLE IF EXISTS allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id1 UInt32, id2 UInt32) ENGINE = MergeTree() ORDER BY (id1 + 1, id2, id1 + 1);  -- { serverError BAD_ARGUMENTS }
+CREATE TABLE allow_suspicious_indices (id1 UInt32, id2 UInt32) ENGINE = MergeTree() ORDER BY (id1 + 1, id2, id1 + 1) SETTINGS allow_suspicious_indices = 1;
+
+DROP TABLE allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id1 UInt32, id2 UInt32, INDEX idx (id1 + 1, id2, id1 + 1) TYPE minmax) ENGINE = MergeTree() ORDER BY id1;  -- { serverError BAD_ARGUMENTS }
+CREATE TABLE allow_suspicious_indices (id1 UInt32, id2 UInt32, INDEX idx (id1 + 1, id2, id1 + 1) TYPE minmax) ENGINE = MergeTree() ORDER BY id1 SETTINGS allow_suspicious_indices = 1;
+
+DROP TABLE allow_suspicious_indices;
+CREATE TABLE allow_suspicious_indices (id1 UInt32) ENGINE = MergeTree() ORDER BY id1 + 1;
+ALTER TABLE allow_suspicious_indices ADD COLUMN id2 UInt32, MODIFY ORDER BY (id1 + 1, id2 + 1, id2 + 1);  -- { serverError BAD_ARGUMENTS }
+ALTER TABLE allow_suspicious_indices ADD COLUMN id2 UInt32, MODIFY ORDER BY (id1 + 1, id2 + 1, id1 + 1);  -- { serverError BAD_ARGUMENTS }
+ALTER TABLE allow_suspicious_indices ADD INDEX idx (id1 + 1, id1 + 1) TYPE minmax;  -- { serverError BAD_ARGUMENTS }
+
+DROP TABLE allow_suspicious_indices;
```

Comment:
@lingtaolf There is still some weird behavior with `ALTER TABLE` and `allow_suspicious_indices = 1` (i.e. in l. 15ff). Let me explain:

First, I wonder why `allow_suspicious_indices` is provided as table parameter in l. 16. It should really be
- a session parameter (syntax: `SET allow_suspicious_indices = 1` and thus apply to all following SQL statements of the session) OR
- provided to `ALTER TABLE` (l. 17)

When I do the latter, i.e.

```sql                                                                         
CREATE TABLE allow_suspicious_indices (id1 UInt32) ENGINE = MergeTree() ORDER BY id1 + 1;                               
ALTER TABLE allow_suspicious_indices ADD INDEX idx (id1 + 1, id1 + 1) TYPE minmax SETTINGS allow_suspicious_indices = 1;
```

the setting doesn't take effect - I receive an exception that the index contains duplicate expressions. Maybe it is because ALTER TABLE is a composite statement for multiple operations and that provided settings are somehow not propagated properly.

Unfortunately, the problem persists also with the first syntax:

```sql
SET allow_suspicious_indices = 1;                                                        
CREATE TABLE allow_suspicious_indices (id1 UInt32) ENGINE = MergeTree() ORDER BY id1 + 1;
ALTER TABLE allow_suspicious_indices ADD INDEX idx (id1 + 1, id1 + 1) TYPE minmax; -- unexpectedly throws      
```

Would you like to check again? Thanks a lot!

Reply:
Checking further, the following behavior is also unexpected:

```sql
CREATE TABLE allow_suspicious_indices (id1 UInt32, id2 UInt32) ENGINE = MergeTree() ORDER BY (id1 + 1, id2, id1 + 1);  -- as expected: throws an exception because that's the new default behavior

SET allow_suspicious_indices = 1;                                                                                                                      
CREATE TABLE allow_suspicious_indices (id1 UInt32, id2 UInt32) ENGINE = MergeTree() ORDER BY (id1 + 1, id2, id1 + 1) ; -- unexpectedly throws an exception ... the settings don't take effect
```

Reply:
ok, I will check it

Reply:
Updated

Reply:
> 

If user specified this parameter via `SET` and `SETTINGS` at same time , which one will works?

Reply:
Table Settings seem can't be set by session parameter while creating a table.

Reply:
"Local" settings (= per-query settings via `SETTINGS`) take precedence before "global" settings (e.g. session settings via `SET`).

Reply:
<img width="1139" alt="image" src="https://user-images.githubusercontent.com/87509571/233613593-bc1f682f-dd0b-47a9-899a-061bf13b31ea.png">
I've tested this behavior with clickhouse. The session parameters can't work on DDL. I've set `parts_to_throw_insert` via session parameter and create a table. After that, I can't see this settings in table DDL.
So, should we process this new setting like  `parts_to_throw_insert `?

Reply:
Nevermind. I had a closer look at the logic again and it works fine (i.e. consistent and as intuitively expected). I adjusted the test a bit. Once CI runs are green, I'll merge.